lib/debugger.coffee

```diff
@@ -23,19 +24,29 @@ class ProcessManager extends EventEmitter
     result[key(e)] = value(e) for e in env.split(";")
     return result
 
-  start: (file) ->
+  startActiveFile: () ->
+    @start true
+
+  start: (withActiveFile) ->
+    startActive = withActiveFile
     @cleanup()
       .then =>
+        packagePath = @atom.project.resolvePath('package.json')
+        packageJSON = require packagePath if fs.existsSync(packagePath)
```

Comment:
@kiddkai re:
```Instead of using resolvePath, is that better to do something like find the first package.json from current .js file? Assuming is will be a atom project looks like(I saw a lot of people do this):

atom-project/
  -- app1/
  | -- package.json
  | -- lib/
    | -- something.js
  -- app2/
  | -- package.json
  | -- lib/
    | -- app.js```

*sorry, i found/fixed a bug, and it outdated the diff

that's interesting.  I get the usefulness of that.  But i'm not sure the debugger should make that assumption.

You're right.  Assuming the project is structured that way- it makes sense to try and debug that specific package/app.
But i think coupling the functionality of the debugger to that specific file structure maybe a bit too prescriptive.

What if the `atom-project/` folder above- had it's own `package.json` in the root?  Would you stop at the first `package.json`, as you walk the directories upward?
What if the open file in the editor is a completely unrelated file?  Like the notes file on my desktop.

To take a step back a bit-  I think it's an important feature when it comes to having a multi project debugging environment.  And something i've actually been thinking about taking a stab at.

I also think the first step in that direction- might need to be a way to support multiple debug configurations in a single atom session.  Or the assumption would be that all the projects in the editor, would be configured/structured exactly the same way.
I'm not sure that's a great stipulation for debugging multiple projects in the same editor session.

in related news- you made me test something and i fixed a bug. :smile: